added docs for the prev auction info module

jlquaccia opened the pull request:
- Documentation for the new Previous Auction Info module

## 🏷 Type of documentation
- [x] new feature

## 📋 Checklist
- [x] Related pull requests in prebid.js or server are linked -> Paste link in this list or reference it on the PR itself

Relative links
- PR: https://github.com/prebid/Prebid.js/pull/11086
- GH Issue: https://github.com/prebid/Prebid.js/issues/10788

pm-harshad-mane commented:
@jlquaccia can you pease work on getting this PR merged?

jlquaccia commented:
Fixed some linting errors that were present and also rebased on top of master.  @ChrisHuie @dgirardi could one of you guys review when you have a chance?

dgirardi commented on dev-docs/publisher-api-reference/setConfig.md:
```diff
@@ -1404,6 +1404,63 @@ pbjs.setConfig({
+    prebid: {
+      previousauctioninfo: [
+        {
+          bidderRequestId: "123abc",
+          bidId: "456def",
+          rendered: 1,
+          source: "pbjs",
+          adUnitCode: "div-gpt-ad-123-0",
+          highestTargetedBidCpm: "0.05",
+          targetedBidCpm: "0.04",
+          highestBidCpm: 0.052275935,
+          bidderCpm: 0.04,
+          bidderOriginalCpm: 0.04,
+          bidderCurrency: "USD",
+          bidderOriginalCurrency: "USD",
+          bidderErrorCode: -1,
+          timestamp: 1739400860310
```
these may need an explanation, especially if we keep the current "null" values. Here's an example (from https://github.com/prebid/Prebid.js/issues/12825)

```json
{
  "bidderRequestId": "58285d8b5b2d6c8",
  "bidId": "66630bf7bfef2e",
  "rendered": 0,
  "source": "pbjs",
  "adUnitCode": "/19968336/header-bid-tag-1",
  "highestTargetedBidCpm": "",
  "targetedBidCpm": "",
  "highestBidCpm": 0,
  "bidderCpm": "nobid",
  "bidderOriginalCpm": "nobid",
  "bidderCurrency": "nobid",
  "bidderOriginalCurrency": "nobid",
  "bidderErrorCode": -1,
  "timestamp": 1741037479568
}
```

from this you can see the null  value can be `0`, `-1`, `''`, `'nobid'` depending on the field. I proposed in https://github.com/prebid/Prebid.js/issues/12822 to just use `null`, which we're still early enough to do; either way it'd help to elaborate on what these fields mean.

dgirardi commented on dev-docs/publisher-api-reference/setConfig.md:
````diff
@@ -1404,6 +1404,63 @@ pbjs.setConfig({
+
+```javascript
+import { enablePreviousAuctionInfo } from '../libraries/previousAuctionInfo/previousAuctionInfo.js';
+enablePreviousAuctionInfo({ bidderCode: 'some bidder code' });
````
see https://github.com/prebid/Prebid.js/issues/12822 - this approach is not friendly to PBS and we may want to change it.

bretg commented on dev-docs/publisher-api-reference/setConfig.md:
```diff
@@ -1404,6 +1404,63 @@ pbjs.setConfig({
+1. A Prebid.js auction runs and completes
+1. At the end of an auction, details about the auction are collected from each bidder using the module
+1. If a Prebid bid wins, then the `rendered` field is updated to `1` to indicate this in the collected auction data for all bidders who bid on the same adunit
+1. During the next Prebid.js auction, if a bidder has this module enabled AND submits a valid bid request, then previous auction info data will be injected into the bid request of the new auction within the following path: `ortb2.ext.prebid.previousAuctionInfo`
```
Please confirm the ORTB path -- this line says `ext.prebid.previousAuctionInfo` (camel case) while the example below is all lower case.

We do prefer the lowercase for the ORTB protocol.

jlquaccia commented:
@dgirardi @bretg pushed a new commit addressing your comments above.  The changes in the new commit relate to and depend on the changes in https://github.com/prebid/Prebid.js/pull/12831 which were proposed on https://github.com/prebid/Prebid.js/issues/12822.

However it seems at the moment, that there is some discussion on https://github.com/prebid/Prebid.js/issues/12822 in terms of how exactly to proceed..

dgirardi commented on dev-docs/modules/previousAuctionInfo.md:
```diff
@@ -0,0 +1,76 @@
+1. A Prebid.js auction runs and completes.
+1. At the end of an auction, details about the auction are collected from each bidder using the module.
+1. If a Prebid bid wins, then the `rendered` field is updated to `1` to indicate this in the collected auction data for all bidders who bid on the same adunit within the same Prebid auction.
+1. During the next Prebid.js auction, if a bidder has this module enabled AND submits a valid bid request, then previous auction info data will be injected into the bidder's bid request of the new auction within the following path: `ortb2.ext.prebid.previousauctioninfo`.
```
if a bidder has this module enabled -> if a bidder is included in `bidders` (bidders no longer enable themselves)
submits a valid bid request -> is included in the auction (bidders do not submit bid requests, unless you're referring to those that go to their SSPs)

dgirardi commented on dev-docs/publisher-api-reference/setConfig.md:
```diff
@@ -1404,6 +1404,18 @@ pbjs.setConfig({
 
 <a name="setConfig-Generic-Configuration"></a>
 
+### Enable Previous Auction Info
```
typically configuration that requires a particular module is not listed here but on that module's page.
